### PR TITLE
Allow printing custom types with their values

### PR DESCRIPTION
**Pull request by segiddins**
this now works

```swift
indirect type Foo {
  let x: Int
  var y: Foo
  let z: String
}

type Bar {
  let x: Int
  let z: String
}

func fhdjsfhjskdhfjkds() {
  let x = 5 as Any
  println(x as Int)
}

func nooooo() {
  let x = Bar(x: 5, z: String(cString: "fdshfjkdshfksd")) as Any
  println(x)
}

func main() {
  nooooo()
  fhdjsfhjskdhfjkds()
  let x = ("a", 4) //as Any

  println(x)
  let anyX = x as Any

  let foo = Foo(x: 0, y:
      Foo(x: 6540, y: nil, z: String(cString: "fdshfjkdshfksd")),
    z: String(cString: "fdsfs"))

  println(((foo.y as Any) as Foo).x)
  println(foo)
}
```

### CODE REVIEW

File: Sources/IRGen/TypeIRGen.swift [hunk 1]
```diff
@@ -72,11 +72,17 @@ extension IRGenerator {
       }
       fields = decl.fields.map { ($0.name.name, $0.type) }
     case .tuple(let types):
-      fields = types.map { (nil, $0) }
+      fields = types.enumerated().map { ($0.offset.description, $0.element) }
```
Comment (harlanhaskins) [inline Sources/IRGen/TypeIRGen.swift]: Should this offset description be prefaced with a "."? `".\($0.offset)"`?

File: Sources/Import/ClangImporter.swift [hunk 2]
```diff
@@ -386,9 +384,41 @@ class ClangImporter: Pass {
     
     let name = clang_getTokenSpelling(tu, tokens[0]).asSwift()
     guard context.global(named: name) == nil else { return }
-    guard clang_getTokenKind(tokens[1]) == CXToken_Literal else { return }
-    guard let assign = parse(tu: tu, token: tokens[1], name: name) else { return }
-    context.add(assign)
+    // TODO: allow importing identifiers
```
Comment (harlanhaskins) [inline Sources/Import/ClangImporter.swift]: TODO is no longer necessary 😅

File: Sources/Runtime/metadata.cpp [hunk 3]
```diff
@@ -210,5 +222,18 @@ void *trill_checkedCast(void *anyValue_, void *typeMetadata_) {
...
   return trill_getAnyValuePtr(anyValue_);
 }
-  
+
+uint8_t trill_anyIsNil(TRILL_ANY any_) {
+  auto any = reinterpret_cast<AnyHeader*>(any_._any);
+  trill_assert(any != nullptr);
+  auto metadata = reinterpret_cast<TypeMetadata*>(any->typeMetadata);
+  auto pointerLevel = metadata->pointerLevel;
+  if (pointerLevel > 0) { return 0; }
```
Comment (harlanhaskins) [inline Sources/Runtime/metadata.cpp]: Should this be `== 0`?

**Review by harlanhaskins — COMMENTED** [review]
LGTM modulo comments

**Comment by harlanhaskins** [comment]
Also you'll need to fix the conflicts

**Comment by segiddins** [comment]
Ugh this looks like a nasty rebase

**Comment by harlanhaskins** [comment]
😬 if you want I can take a spin at it

**Comment by segiddins** [comment]
Rebased

**Review by harlanhaskins — APPROVED** [review]
Rebase looks great! I'm excited to merge this!

**Pull request merged**